src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -30,12 +30,32 @@ static ucs_config_field_t uct_cuda_copy_md_config_table[] = {
 
 static ucs_status_t uct_cuda_copy_md_query(uct_md_h md, uct_md_attr_t *md_attr)
 {
-    md_attr->cap.flags            = UCT_MD_FLAG_REG;
+    ucs_status_t status;
+    int          active;
+    size_t       total;
+    CUdevice     dev;
+
+    UCT_CUDADRV_CTX_ACTIVE(active);
+    if (!active) {
+        total = 0;
```

Comment:
@yosefe @bureddy From the UCP point of view, do you see an issue in md_query returning available memory as zero if cuda device context hasn't been set? At the point of needing to use device bounce buffer pools for tag operations with cuda buffers, there should be a context available and the goal is to avoid allocating on device 0 in the visible device list by default.